main/src/library/List.flix

```diff
@@ -1103,7 +1103,7 @@ namespace List {
     ///
     /// The sort implementation is a mergesort.
     ///
-    pub def sort(xs: List[a]): List[a] with Order[a] = sortWith(Order.compare, xs)
+    pub def sort(xs: List[a]): List[a] & Impure with Order[a] = Array.toList(Array.sort(toArray(xs)))
```

Comment:
Instead of having an Impure effect, you should cast the result to pure. (because in this case it is actually pure).

That is `Array.toList(Array.sort(toArray(xs))) as & Pure`.

Same comment applies below.